logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/MessageContent.kt

```diff
@@ -1,18 +1,30 @@
 package com.wire.kalium.logic.data.message
 
+import com.wire.kalium.logic.data.conversation.Member
 import com.wire.kalium.protobuf.messages.QualifiedConversationId
 
 sealed class MessageContent {
-    data class Text(val value: String) : MessageContent()
-    data class Calling(val value: String) : MessageContent()
-    data class Asset(val value: AssetContent) : MessageContent()
-    data class DeleteMessage(val messageId: String) : MessageContent()
-    data class TextEdited(val editMessageId :String, val newContent : String) : MessageContent()
+
+    sealed class Client : MessageContent()
+    sealed class Server : MessageContent()
+
+    // client message content types
+    data class Text(val value: String) : Client()
+    data class Calling(val value: String) : Client()
+    data class Asset(val value: AssetContent) : Client()
+    data class DeleteMessage(val messageId: String) : Client()
+    data class TextEdited(val editMessageId :String, val newContent : String) : Client()
     data class DeleteForMe(
         val messageId: String,
         val conversationId: String,
         val qualifiedConversationId: QualifiedConversationId?
-    ) : MessageContent()
+    ) : Client()
+
+    object Unknown : Client()
 
-    object Unknown : MessageContent()
+    // server message content types
+    sealed class MemberChange(open val members: List<Member>) : Server() {
+        data class Join(override val members: List<Member>) : MemberChange(members)
+        data class Leave(override val members: List<Member>) : MemberChange(members)
```

Comment:
Maybe rename to past tense to indicate that event already occurred?
```suggestion
        data class Joined(override val members: List<Member>) : MemberChange(members)
        data class Left(override val members: List<Member>) : MemberChange(members)
```

Reply:
Sure, but for event names I kept the current ones because "member-leave" and "member-join" are the exact types so it's easier to find and understand it, but here it can be changed, but I prefer "Added"/"Removed" because "Joined"/"Left" suggest that user did it himself and actually it covers both cases ("left" only if author is the only member on list)

Reply:
yeah, there is no a name to rule them all, so I would leave it to your own personal preference then 🧑‍🎨